### PR TITLE
Use path instead of absolute_path for inside eval

### PR DESCRIPTION
**Pull request by HParker**
This prevents getting a nil absolute_path for code executed inside of an eval

### Summary

Absolute path is not always available on your callers, but path is more likely to be available. This especially fixes files that are evaluated which in ruby 3.1 never have absolute paths after this change in ruby: https://github.com/ruby/ruby/commit/64ac984129a7a4645efe5ac57c168ef880b479b2

### CODE REVIEW

**Review by BlakeWilliams — COMMENTED** [review]
This looks great!

Would you mind adding a test and a changelog entry for this change? 🙇

**Comment by HParker** [comment]
Thanks @BlakeWilliams updated!

**Comment by HParker** [comment]
oh blah changelog conflict, one sec...

File: test/view_component/base_test.rb [hunk 1]
```diff
@@ -90,4 +90,20 @@ def test_does_not_render_additional_newline_with_render_in
...
+          @cta = cta
+          @title = title
+        end
+        private
+
+        attr_reader :cta, :title
+      end
+    RUBY
+
+    eval(source)
```
Comment (joelhawksley) [inline test/view_component/base_test.rb:107]: Looks like we need to add a rubocop exception for this line for our linters to pass ❤️

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ title: Changelog
 
 ## main
 
+* Fix path handling for evaluated view components
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:10]:
```suggestion
* Fix path handling for evaluated view components that broke in Ruby 3.1.
```

File: test/view_component/base_test.rb (hunk 1 above)
Comment (joelhawksley) [inline test/view_component/base_test.rb:107]:
```suggestion
    eval(source) # rubocop:disable Security/Eval
```

**Review by joelhawksley — APPROVED** [review]

**Pull request merged**